### PR TITLE
feat(hub-common): add facade methods related to entity sharing and ac…

### PR DESCRIPTION
**Pull request by rweber-esri**
…cess in support of extending co

affects: @esri/hub-common

ISSUES CLOSED: 8473

1. Description:

Adds facade methods related to sharing and access to extend collaborators workspace support for events 3

1. Instructions for testing:

1. Closes Issues: [#8473](https://devtopia.esri.com/dc/hub/issues/8473)

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

**Comment by rweber-esri** [comment]
This is ready for review @juliannaeapicella & @dbouwman. I've tested these changes extensively against my [opendata-ui PR-13385](https://github.com/ArcGIS/opendata-ui/pull/13385) (`?uiVersion=PR-13385`). Specifically:
* changing access works as expected across all entities (tested Groups, Initiatives, Projects & Events workspaces)
* managing groups works as expected across all entity workspaces that have collaborators panes (tested Initiatives, Projects & Events)

File: packages/common/src/events/_internal/shareEventWithGroups.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { poll } from "../../utils/poll";
+
+/**
+ * Shares an event with one or more groups
+ * @param groupIds An array of group IDs to share the event with
+ * @param entity An IHubEvent object
+ * @param context An IArcGISContext object
+ * @returns a promise that resolves IHubItemEntity
+ */
+export async function shareEventWithGroups(
```
Comment (dbouwman) [inline packages/common/src/events/_internal/shareEventWithGroups.ts:15]: consider adding more comments about what's going on in this - I'm not following why we're polling a search for groups etc, so my guess is that others on the team, including "Future Randy" will also be a little O_o when they need to get into this code :)

Comment (rweber-esri) [reply packages/common/src/events/_internal/shareEventWithGroups.ts:15]: Will do. For context, _new_ groups aren't _immediately_ available in the AGO search index. This prevents the need for us to add polling potentially in many places of the app where we create new groups and then want to immediately share an item to that group. In 99% of cases, this should only fire a single request. The collaborators pane had implemented this polling before sharing the entity with 1 or more groups, but had a bug where `num` wasn't specified on the polling request. This was problematic in the scenario where the user selected 11+ groups to share the entity with. The validation function expected 11 results, but AGO search would only return 10 (the default num of results when `num` param is omitted).

File: packages/common/src/utils/poll.ts [hunk 2]
```diff
@@ -11,14 +11,16 @@
  * @param validationFn
  * @param opts
  */
```
Comment (dbouwman) [inline packages/common/src/utils/poll.ts:13]: ❤️❤️❤️❤️ Only as is that the polling also implement an incremental back-off - so if it's not valid after 300ms, the next timeout is 600ms. This is what AGO asked us to do if we use a polling strategy b/c the api is not deterministic

Comment (rweber-esri) [reply packages/common/src/utils/poll.ts:13]: I thought about this too. Will push up a quick change to add incremental backoff

**Review by dbouwman — APPROVED** [review]
This looks great! Thanks for adding the polling - only ask is to add "incremental back-off" to `poll` and add a little more description of what's happening in the share/unshare functions

**Pull request merged**